ibm/resource_ibm_schematics_job.go

```diff
@@ -0,0 +1,2040 @@
+			Type:                       TypeString,
+			Optional:                   true,
+			AllowedValues:              "ansible_playbook_check, ansible_playbook_run, helm_install, helm_list, helm_show, opa_evaluate, terraform_init, terrform_apply, terrform_destroy, terrform_plan, terrform_refresh, terrform_show, terrform_taint, workspace_apply_flow, workspace_custom_flow, workspace_destroy_flow, workspace_init_flow, workspace_plan_flow, workspace_refresh_flow, workspace_show_flow",
+		},
+		ValidateSchema{
+			Identifier:                 "location",
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Optional:                   true,
+			AllowedValues:              "eu-de, eu-gb, us-east, us-south",
+		})
+
+	resourceValidator := ResourceValidator{ResourceName: "ibm_schematics_job", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIBMSchematicsJobCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	schematicsClient, err := meta.(ClientSession).SchematicsV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	session, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	iamRefreshToken := session.Config.IAMRefreshToken
+
+	createJobOptions := &schematicsv1.CreateJobOptions{}
+	createJobOptions.SetRefreshToken(iamRefreshToken)
+
+	if _, ok := d.GetOk("command_object"); ok {
+		createJobOptions.SetCommandObject(d.Get("command_object").(string))
+	}
+	if _, ok := d.GetOk("command_object_id"); ok {
+		createJobOptions.SetCommandObjectID(d.Get("command_object_id").(string))
+	}
+	if _, ok := d.GetOk("command_name"); ok {
+		createJobOptions.SetCommandName(d.Get("command_name").(string))
+	}
+	if _, ok := d.GetOk("command_parameter"); ok {
+		createJobOptions.SetCommandParameter(d.Get("command_parameter").(string))
+	}
+	if _, ok := d.GetOk("command_options"); ok {
+		createJobOptions.SetCommandOptions(d.Get("command_options").([]string))
+	}
+	if _, ok := d.GetOk("job_inputs"); ok {
+		var inputs []schematicsv1.VariableData
+		for _, e := range d.Get("job_inputs").([]interface{}) {
+			value := e.(map[string]interface{})
+			inputsItem := resourceIBMSchematicsJobMapToVariableData(value)
+			inputs = append(inputs, inputsItem)
+		}
+		createJobOptions.SetInputs(inputs)
+	}
+	if _, ok := d.GetOk("job_env_settings"); ok {
+		var settings []schematicsv1.VariableData
+		for _, e := range d.Get("job_env_settings").([]interface{}) {
+			value := e.(map[string]interface{})
+			settingsItem := resourceIBMSchematicsJobMapToVariableData(value)
+			settings = append(settings, settingsItem)
+		}
+		createJobOptions.SetSettings(settings)
+	}
+	if _, ok := d.GetOk("tags"); ok {
+		createJobOptions.SetTags(expandStringList(d.Get("tags").([]interface{})))
+	}
+	if _, ok := d.GetOk("location"); ok {
+		createJobOptions.SetLocation(d.Get("location").(string))
+	}
+	if _, ok := d.GetOk("status"); ok {
```

Comment:
you are directly accessing status.0 with out checking any length...suppose if user defines in hos tempalte 
```
state {
}
```
Then it will crash

Reply:
Done. Updated this code and many other places to check length of the array before accessing